client/landing/gutenboarding/onboarding-block/signup-form/index.tsx

```diff
@@ -0,0 +1,68 @@
+/**
+ * External dependencies
+ */
+import React, { useState } from 'react';
+
+// /**
+//  * Internal dependencies
+//  */
+import {
+	usePasswordlessSignUp,
+	UsePasswordlessSignUpStatus,
+	Provider,
+} from '@automattic/authentication';
+import { Button, TextControl, Modal } from '@wordpress/components';
+
+import './style.scss';
+
+const SignupForm = () => {
+	const [ emailVal, setEmailVal ] = useState( '' );
+	const { signUp, status, error } = usePasswordlessSignUp();
+
+	const handleSignUp = ( event: React.FormEvent< HTMLFormElement > ) => {
+		event.preventDefault();
+		signUp( emailVal );
+	};
+
+	const renderTosLink = () => {
+		return (
+			<a href="https://wordpress.com/tos/" target="_blank" rel="noopener noreferrer">
+				Terms of Service
+			</a>
+		);
+	};
+
+	return (
+		<Modal title="Sign up to save your changes" onRequestClose={ () => {} }>
+			<form onSubmit={ handleSignUp }>
+				<label htmlFor="email">Your Email Address</label>
+				<TextControl
+					id="email"
+					value={ emailVal }
+					onChange={ setEmailVal }
+					placeholder="yourname@email.com"
+				/>
+				<p>By creating an account you agree to our { renderTosLink() }.</p>
```

Comment:
This one will be a fun one to try apply `__experimentalCreateInterpolateElement` from `@wordpress/i18n` :-)

Reply:
It turns out that the` __experimentalCreateInterpolateElement` in `@wordpress/i18n` isn't in the version of the package that we're using (3.4.0). We had that card to update the packages to the latest version but I couldn't find it using the issue search but I think we will have to do that first.

Reply:
>We had that card to update the packages to the latest version but I couldn't find it using the issue search but I think we will have to do that first.

This was the latest I saw go in https://github.com/Automattic/wp-calypso/pull/38554

Reply:
It might make sense to just format this paragraph differently for now to get this merged and update the copy later.

Reply:
`__experimentalCreateInterpolateElement` is available now :) but it's not in the typings :(

It works well so I'll stick with it, but I'll look for some way to hack it in to the types of our project before submitting something to the official repo.